lib/active_merchant/billing/gateways/blue_snap.rb

```diff
@@ -433,15 +433,15 @@ def parse_element(parsed, node)
         end
       end
 
-      def api_request(action, request, verb, payment_method_details)
-        ssl_request(verb, url(action, payment_method_details), request, headers)
+      def api_request(action, request, verb, payment_method_details, options)
```

Comment:
It's kind of weird to me that we have this `api_request` method wrapping the `ssl_request` method. We could just calculate the `url` and the `header` inside the `commit` method and then call `ssl_request` directly [the way we do in most other gateway adapters](https://github.com/activemerchant/active_merchant/blob/master/lib/active_merchant/billing/gateways/adyen.rb#L557).

This is just an idle thought on my part, not a request for changes. It has nothing to do with the changes you are making in this PR.